scripts/build.py

```diff
@@ -322,48 +420,61 @@ def get_dependencies_filepaths():
-
-    dependency_filepaths = get_dependencies_filepaths()
-    _minify_and_create_sourcemap(dependency_filepaths['js'], third_party_js)
-    _join_files(dependency_filepaths['css'], third_party_css)
-    _minify(third_party_css, third_party_css)
-    _copy_fonts(dependency_filepaths['fonts'], fonts_dir)
-
-
-def build_third_party_libs():
+    THIRD_PARTY_JS_FILEPATH = os.path.join(
+        third_party_directory_path, 'js', 'third_party.js')
+    MINIFIED_THIRD_PARTY_JS_FILEPATH = os.path.join(
+        third_party_directory_path, 'js', 'third_party.min.js')
+    THIRD_PARTY_CSS_FILEPATH = os.path.join(
+        third_party_directory_path, 'css', 'third_party.css')
+    MINIFIED_THIRD_PARTY_CSS_FILEPATH = os.path.join(
+        third_party_directory_path, 'css', 'third_party.min.css')
+
+    _minify_and_create_sourcemap(
+        THIRD_PARTY_JS_FILEPATH, MINIFIED_THIRD_PARTY_JS_FILEPATH)
+    _minify(THIRD_PARTY_CSS_FILEPATH, MINIFIED_THIRD_PARTY_CSS_FILEPATH)
+    # Clean up un-minified third_party.js and third_party.css.
+    os.remove(THIRD_PARTY_JS_FILEPATH)
+    os.remove(THIRD_PARTY_CSS_FILEPATH)
+
+
+def build_third_party_libs(third_party_directory_path):
     """Joins all third party css files into single css file and js files into
     single js file. Copies both files and all fonts into third party folder.
     """
 
-    print 'Building third party libs...'
+    print 'Building third party libs at %s' % third_party_directory_path
+
+    ensure_directory_exists(third_party_directory_path)
+    # Regenerate /third_party/generated from scratch.
+    shutil.rmtree(third_party_directory_path)
 
-    third_party_js = os.path.join(
-        THIRD_PARTY_GENERATED_DEV_DIR, 'js', 'third_party.js')
-    third_party_css = os.path.join(
-        THIRD_PARTY_GENERATED_DEV_DIR, 'css', 'third_party.css')
-    fonts_dir = os.path.join(THIRD_PARTY_GENERATED_DEV_DIR, 'fonts', '')
+    THIRD_PARTY_JS_FILEPATH = os.path.join(
+        third_party_directory_path, 'js', 'third_party.js')
+    THIRD_PARTY_CSS_FILEPATH = os.path.join(
+        third_party_directory_path, 'css', 'third_party.css')
+    FONTS_DIR = os.path.join(third_party_directory_path, 'fonts', '')
 
     dependency_filepaths = get_dependencies_filepaths()
-    _join_files(dependency_filepaths['js'], third_party_js)
-    _join_files(dependency_filepaths['css'], third_party_css)
-    _copy_fonts(dependency_filepaths['fonts'], fonts_dir)
+    ensure_directory_exists(THIRD_PARTY_JS_FILEPATH)
+    with open(THIRD_PARTY_JS_FILEPATH, 'w+') as third_party_js_file:
+        _join_files(dependency_filepaths['js'], third_party_js_file)
+
+    ensure_directory_exists(THIRD_PARTY_CSS_FILEPATH)
+    with open(THIRD_PARTY_CSS_FILEPATH, 'w+') as third_party_css_file:
+        _join_files(dependency_filepaths['css'], third_party_css_file)
+
+    ensure_directory_exists(FONTS_DIR)
+    copy_tasks = collections.deque()
+    copy_tasks += _generate_copy_tasks_for_fonts(
```

Comment:
You could remove the previous line and just use `copy_tasks =` here, right? (Or even inline the whole thing in line 477.)

Reply:
You are absolutely correct! Inlined creating `copy_tasks` and `delete_tasks`!